app/src/main/java/com/zarinraim/dogsearch/presentation/main/all_breeds/BreedsListScreen.kt

```diff
@@ -61,30 +67,30 @@ fun BreedsList(
         verticalArrangement = Arrangement.spacedBy(8.dp),
         state = listState
     ) {
-        items(mainBreedsList) { dogBreed ->
+
+        items(mainBreedsList) { breed ->
             BreedItem(
-                breedName = dogBreed,
-                subBreeds = breeds[dogBreed],
-                navController = navController
+                breed = breed,
+                subBreeds = breeds.map[breed],
+                onClickOpenImage = onClickOpenImage
             )
         }
     }
 }
 
-
 @ExperimentalMaterialApi
 @Composable
 fun BreedItem(
-    breedName: String,
-    subBreeds: List<String>?,
-    navController: NavController
+    breed: Breed,
+    subBreeds: SubBreeds?,
+    onClickOpenImage: (String, String) -> Unit
 ) {
     val expanded = remember { mutableStateOf(false) }
     val rotationState by animateFloatAsState(
         targetValue = if (expanded.value) 180f else 0f
     )
 
-    val hasSubBreeds = subBreeds!!.isNotEmpty()
+    val hasSubBreeds = subBreeds!!.list.isNotEmpty()
```

Comment:
You shouldn't use the not-null assertion operator, please try to handle somehow the situation if subBreeds is null.

Reply:
Changed to assigning default `false` value for this variable.